### PR TITLE
Use `.select()` for entrypoints under `importlib_metadata>=5.0.0`

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
importlib_metadata>=5.0.0 deprecates the dict structure of the entrypoints object, so we use `.select()` instead for iterating over them if needed.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by whytewolf — DISMISSED** [review]

File: salt/utils/entrypoints.py [hunk 1]
```diff
@@ -38,10 +38,18 @@ def iter_entry_points(group, name=None):
     entry_points_listing = []
     entry_points = importlib_metadata.entry_points()
 
-    for entry_point_group, entry_points_list in entry_points.items():
-        if entry_point_group != group:
-            continue
-        for entry_point in entry_points_list:
+    # pre importlib-metadata 5.0.0
```
Comment (s0undt3ch) [inline salt/utils/entrypoints.py:41]: try/catch is faster than `hasattr` and let's account for the newer way first, ie:
```python
def iter_entry_points(group, name=None):
    entry_points_listing = []
    entry_points = importlib_metadata.entry_points()

    try:
        for entry_point in entry_points.select(group=group):
            if name is not None and entry_point.name != name:
                continue
            entry_points_listing.append(entry_point)
    except AttributeError:
        # importlib-metadata < 5.0.0
        for entry_point_group, entry_points_list in entry_points.items():
            if entry_point_group != group:
                continue
            for entry_point in entry_points_list:
                if name is not None and entry_point.name != name:
                    continue
                entry_points_listing.append(entry_point)

    return entry_points_listing
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
We should probably also do a followup PR where we update the version of importlib_metadata we vendor in `salt/ext/`.

**Review by s0undt3ch — APPROVED** [review]

**Comment by akshaypurohit** [comment]
@MKLeb @s0undt3ch  Are there plans to merge this change to previous versions as well?

**Comment by MKLeb** [comment]
@akshaypurohit I would say it is unlikley, as the previous major versions are all past Phase 2 support and have entered the CVE only phase. (https://docs.saltproject.io/salt/install-guide/en/latest/topics/salt-version-support-lifecycle.html)

**Pull request merged**